compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/optimizations/PropertyAccessorInlineLowering.kt

```diff
@@ -46,13 +48,25 @@ class PropertyAccessorInlineLowering(private val context: CommonBackendContext)
             }
             if (property.isEffectivelyExternal()) return expression
 
+            val backingField = property.backingField ?: return expression
+
             if (property.isConst) {
                 val initializer =
-                    (property.backingField?.initializer ?: error("Constant property has to have a backing field with initializer"))
-                return initializer.expression.deepCopyWithSymbols()
+                    (backingField.initializer ?: error("Constant property has to have a backing field with initializer"))
+                val constExpression = initializer.expression.deepCopyWithSymbols()
+                val receiver = expression.dispatchReceiver
+                if (receiver != null && receiver is IrCall) {
```

Comment:
An interesting question is what should print this program?

```kotlin
object C {
    init {
        println("C::<init>")
    }
    const val x = 42
}

fun main() {
    println("1")
    println(C.x)
    println("2")
    val c = C
    println(c.x)
    println("3")
}
```

Reply:
Now it's
```
1
42
2
C::<init>
42
3
1
D::<init>
43
2
43
3
```

Reply:
@LepilkinaElena it would be nice to reflect current behavior in the box tests.

Reply:
I see now that similar test is already added but only for JVM
https://github.com/JetBrains/kotlin/commit/5be28520fcfa8790e6704b3666ff19aa5f68f65a

If there are some failing cases, may be, it's better to split it?

Reply:
👍
I think, it will be nice to do.

cc @dnpetrov